Amazon.Ion.ObjectMapper.Test/IonObjectSerializerTest.cs

```diff
@@ -135,5 +140,85 @@ public void RespectAnnotationPrefixes()
                 "my.universal.namespace.BussyMcBusface", 
                 new IonSerializer().Serialize(new Bus()));
         }
+
+        [TestMethod]
+        public void DeserializeAnnotatedIonToParentClassNameMatchingAnnotation()
+        {
+            IIonValue ionTruck = valueFactory.NewEmptyStruct();
+            ionTruck.AddTypeAnnotation("Truck");
```

Comment:
Did this not work before you changed the code? How does it know what namespace to look in?

Reply:
No. The previous default behavior would return an object of type Vehicle and not truck.

It searches the loaded namespaces to look to see if there's a matching class name. If there isn't, or it's not assignable to the passed-in type (e.g. a subtype), it will try the passed in type.